src/main/react/app/containers/App/excludedDate.js

```diff
@@ -0,0 +1,367 @@
+import React from "react";
+import DateInput from "./dateInput";
+import { DATE_PRESETS, DATE_TYPE, getDatePresets, MONTHS, ORDERS, PERIODS, WEEKDAYS } from "./constants";
+import {
+  format,
+  formatDate,
+  nextOccurrenceByMonth,
+  nextOccurrenceByYear,
+  nextOccurrenceChineseLunar
+} from "../../utils/date";
+import moment from "moment";
+import _ from 'lodash'
+
+
+export default class ExcludeDate extends React.Component {
```

Comment:
This file is pretty long. Should we consider moving some of this code out into separate files?

Reply:
Yeah, I optimized that by moving one helper function to utils, replacing repeating numbers with lodash.range function in select's options. But still need advices for spliting a file. I'll be much appreciated if I could learn some experience on it.